src/nerve_tools/configure_nerve.py

```diff
@@ -142,11 +143,22 @@ def generate_subconfiguration(
                         'headers': healthcheck_headers,
                     },
                 ],
-                'labels': labels,
+                'labels': default_labels,
             }
 
-            v2_key = '%s.%s:%d.v2.new' % (
-                service_name, zk_location, port,
+            if typ != discover:
```

Comment:
I'd suggest structuring this code slightly different.

Maybe instead of something  like
```
if !X:
    continue

modify_labels

v2_key = blah
if Y:
    write config
```
do something like
```
v2_key = ...
if X and Y:
   modify_labels
   write_config
```